app/scripts/lib/user-agent.js

```diff
@@ -143,6 +143,36 @@ const UserAgent = function (userAgent) {
         minor: parseInt(osVersion[1] || 0, 10),
         patch: parseInt(osVersion[2] || 0, 10)
       };
+    },
+
+    /**
+     * Get the generic operating system name.
+     *
+     * @param {String} [os=this.os.name] Full Operating System name
+     * @returns {String} generic operating system name
+     */
+    genericOSName (os = this.os.name) {
```

Comment:
Noticed this isn't called anywhere in here, guessing it is used in https://github.com/mozilla/fxa-content-server/pull/6404?

Reply:
> Noticed this isn't called anywhere in here, guessing it is used in #6404?

Yup. :D I'm trying to pull out the "extra" parts to make the behemoth slightly easier to review.